sdk/azcore/arm/runtime/policy_bearer_token.go

```diff
@@ -99,3 +109,26 @@ func (b *BearerTokenPolicy) onRequest(req *azpolicy.Request, authNZ func(azpolic
 func (b *BearerTokenPolicy) Do(req *azpolicy.Request) (*http.Response, error) {
 	return b.btp.Do(req)
 }
+
+// parseChallenge parses claims from an authentication challenge so a client can request a token that will satisfy
+// conditional access policies. Returns an empty string when no claims are found or claims aren't in ARM's format.
+// This function isn't universally applicable because RPs differ in challenge format and content.
```

Comment:
What are the implications of this this, does it mean that this might fail in some way for certain RPs?

Reply:
Yes. This function can't parse a Storage challenge, for example. The implication is that we can't simply copy or export this function when another RP adds CAE support.

Reply:
Does it work for all of ARM though?  The fact that it's part of ARM's bearer token policy makes me think yes.

Reply:
It should. At least, that's my assumption given no one on the ARM or Identity side mentioned partial support when we first implemented this a couple years ago.

Reply:
ok I get it now, thanks for explaining.